src/core/lib/security/credentials/tls/grpc_tls_crl_provider.cc

```diff
@@ -53,6 +60,168 @@ std::string IssuerFromCrl(X509_CRL* crl) {
   return ret;
 }
 
+absl::StatusOr<std::shared_ptr<Crl>> ReadCrlFromFile(
+    absl::string_view crl_path) {
+  grpc_slice crl_slice = grpc_empty_slice();
+  grpc_error_handle err = grpc_load_file(crl_path.data(), 1, &crl_slice);
+  if (!err.ok()) {
+    // TODO(gtcooke94) log error differently?
+    gpr_log(GPR_ERROR, "Error reading file %s", err.message().data());
+    grpc_slice_unref(crl_slice);
+    return absl::InvalidArgumentError("Could not load file");
+  }
+  std::string raw_crl = std::string(StringViewFromSlice(crl_slice));
+  absl::StatusOr<std::unique_ptr<Crl>> crl = Crl::Parse(raw_crl);
+  if (!crl.ok()) {
+    grpc_slice_unref(crl_slice);
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Parsing crl string failed with result ", crl.status().ToString()));
+  }
+  grpc_slice_unref(crl_slice);
+  return crl;
+}
+
+struct FileData {
+  char path[MAXPATHLEN];
+  off_t size;
+};
+
+#if defined(GPR_LINUX) || defined(GPR_ANDROID) || defined(GPR_FREEBSD) || \
+    defined(GPR_APPLE)
+void GetAbsoluteFilePath(const char* valid_file_dir,
```

Comment:
Should we use `absl::string_view` over `const char*` for the input parameters, and return a `std::string` rather than have an output parameter?

If we do that, then the logic could be simplified quite a bit, e.g. using `absl::StrFormat`.

Similarly in the Windows implementation of `GetAbsoluteFilePath`.